src/io.jl

```diff
@@ -0,0 +1,145 @@
+                 actually computing the `HybridSystem`; otherwise, instantiate a
+                 `HybridSystem` with the given assumptions
+- `ST`        -- (optional, default: `nothing`) assumption
+                 for the type of mathematical system for each mode
+- `N`         -- (optional, default: `Float64`) numeric type of the system's coefficients
+
+### Output
+
+Hybrid system that corresponds to the given SX model and the given assumptions
+on the system type. If `raw_dict=true`; otherwise, a dictionary with the Julia
+expression objects that define the model. The keys of this dictionary are:
+
+- `automaton`
+- `variables`
+- `transitionlabels`
+- `invariants`
+- `flows`
+- `resetmaps`
+- `switchings`
+
+### Notes
+
+This function makes the following assumptions:
+
+1) The model contains only 1 component. If the model contains more than 1 component,
+   an error is raised. If your model contains more than one component, recall that
+   network components can be flattened using `sspaceex`.
+2) Location identifications ("id" field) are integers.
+4) The default and a custom `ST` parameter assume that all modes are of the same
+   type. In general, you may pass a vector of system's types in `kwargs` (not implemented).
+
+Moreover, let us note that:
+
+1) The tags `<notes> ... <\notes>` are ignored.
+2) Variable names are stored in the dictionary `variables`, together with other
+   information such as if the variable is controlled or not. This dictionary is
+   then stored in the extension field (`ext`) of the hybrid system.
+3) The transition labels are stored in the extension field (`ext`) of the
+   hybrid system.
+4) We use the location "id" field (an integer), such that each of the vectors `modes`,
+   `resetmaps` and `switchings` corresponds to the location with the given "id". For
+   example, `modes[1]` corresponds to the mode for the location with `id="1"`.
+5) The `name` field of a location is ignored.
+
+These comments apply whenever `raw_dict=false`:
+
+1) The `variables` is an ordered dictionary, where the ordered is given by the
+   insertion order. This allows deterministic iteration over the dictionary,
+   (notice that in a usual dictionary, the order in which the elements are returned
+   does not correspond, in general, to the order in which the symbols where saved).
+   The `variables` are stored in the coefficients matrix using this insertion order.
+2) If `ST` is `nothing`, the modes are given as the vector of tuples `(flows, invariants)`,
+   each component being a list of expressions.
+"""
+function readsxmodel(file; raw_dict=false,
+                           ST=nothing,
+                           N=Float64,
+                           kwargs...)
+
+    # 1) Open XML and read number of components and locations
+    # =======================================================
+    sxmodel = readxml(file)
+    root_sxmodel = root(sxmodel)
+
+    lcount, tcount = count_locations_and_transitions(root_sxmodel)
```

Comment:
The variable names are a bit confusing, but I do not see a fundamentally better way.

Reply:
let's se `loc_count` and `trans_count`